feat: upgrade lerna and enable local cache

zahin-mohammad opened the pull request:
- updates from lerna 5 to lerna 8
- enables local nx cache


TICKET: WP-6096

zahin-mohammad committed:
feat: upgrade to lerna 6

TICKET: WP-6096

zahin-mohammad committed:
feat: upgrade to lerna 7

TICKET: WP-6096

zahin-mohammad committed:
feat: upgrade to lerna 8

TICKET: WP-6096

zahin-mohammad committed:
feat: integrate lerna cache

TICKET: WP-6096

zahin-mohammad committed:
chore: add clean-cache command

TICKET: WP-6096

zahin-mohammad committed:
feat: enable cache in CI

TICKET: WP-6096

mukeshsp reviewed (commented):
Please fix the conflicts

zahin-mohammad committed:
feat: skip cache during publish

TICKET: WP-6096

zahin-mohammad committed:
fix(root): resolve nx cache configuration and github actions issues

ISSUE:
- GitHub Actions were failing due to NX cache configuration problems
- Copilot feedback identified redundant exclusions in nx.json production named input
- GitHub Actions cache keys had problematic negation pattern (!modules/*/package-lock.json)

SOLUTION:
- Fix NX cache configuration: remove redundant exclusions in nx.json production named input
- Fix GitHub Actions cache keys: remove problematic negation pattern from all cache keys
- Remove explicit NX dependency (Lerna 8 provides NX integration automatically)

TECHNICAL DETAILS:
- The production named input had redundant exclusions already covered by default input
- GitHub Actions hashFiles() function doesn't handle negation patterns reliably
- Lerna 8 includes NX integration, making explicit nx dependency unnecessary

VERIFICATION:
- NX cache functionality works correctly
- GitHub Actions cache keys function properly
- Cache restoration steps complete successfully in CI

TICKET: WP-6096

zahin-mohammad committed:
fix(babylonlabs-io-btc-staking-ts): resolve build issue caused by dependency conflicts

ISSUE:
- Babylonlabs package (@bitgo/babylonlabs-io-btc-staking-ts) was failing to build
- Root cause: Lerna 8 upgrade introduced newer minimatch versions (9.x-10.x) that
  conflicted with older @types/glob@7.2.0 interface expectations
- Error: Type 'IOptions' is not assignable to type 'IOptions' in dts-bundle-generator

SOLUTION:
- Add @types/glob@^8.1.0 locally to babylonlabs package (compatible with newer minimatch versions)
- Keep dts-bundle-generator at original version ^9.3.1 (no version bump needed)
- Use local dependency approach to keep fix targeted to affected package only

TECHNICAL DETAILS:
- Lerna 8 brings in minimatch@9.x-10.x via NX integration
- @types/glob@7.2.0 expected minimatch@7.x-8.x interface
- @types/glob@8.1.0 is compatible with minimatch@9.x-10.x
- Local dependency prevents affecting other packages in the monorepo

VERIFICATION:
- Babylonlabs package builds successfully from clean cache
- Other packages remain unaffected
- dts-bundle-generator works correctly with updated type definitions

TICKET: WP-6096

zahin-mohammad commented on modules/express/test/unit/clientRoutes/externalSign.ts:
```diff
@@ -100,6 +100,18 @@ describe('External signer', () => {
+  afterEach(async () => {
+    // Clean up the encryptedPrivKeys.json file if it exists
+    try {
+      await fs.promises.unlink('encryptedPrivKeys.json');
```
Out of scope.

zahin-mohammad committed:
refactor(root): extract nx cache key pattern to workflow environment variable

- Add NX_CACHE_FILES environment variable to reduce duplication
- Update all three cache key patterns to use env.NX_CACHE_FILES
- Improves maintainability and reduces copy-paste errors
- Addresses Copilot feedback on cache key pattern duplication

TICKET: WP-6096

zahin-mohammad commented on modules/express/encryptedPrivKeys.json:
```diff
@@ -1,3 +0,0 @@
-{
-   "61f039aad587c2000745c687373e0fa9": "{\"iv\":\"O74H8BBv86GBpoTzjVyzWw==\",\"v\":1,\"iter\":10000,\"ks\":256,\"ts\":64,\"mode\":\"ccm\",\"adata\":\"\",\"cipher\":\"aes\",\"salt\":\"7n8pAjXCfug=\",\"ct\":\"14MjiKBksaaayrwuc/w8vJ5C3yflQ15//dhLiOgYVqjhJJ7iKrcrjtgfLoI3+MKLaKCycNKi6vTs2xs8xJeSm/XhsOE9EfapkfGHdYuf4C6O1whNOyugZ0ZSOA/buDC3rvBbvCNtLDOxN5XWJN/RADOnZdHuVGk=\"}"
-}
```
I updated the test to clean up this file after its done.

zahin-mohammad committed:
chore: express tests should clean up encryptedPrivKeys.json

TICKET: WP-6096
